frontend/src/components/login/InitialLoginStep.tsx

```diff
@@ -31,33 +33,60 @@ export default function InitialLoginStep({
 
     const handleLogin = async () => {
         try {
-          if (!email || !password) {
-            return;
-          }
-    
-          setIsLoading(true);
-          const isLoginSuccessful = await attemptLogin({
-            email,
-            password,
-          });
-          if (isLoginSuccessful && isLoginSuccessful.success) {
-            // case: login was successful
-    
-            if (isLoginSuccessful.mfaEnabled) {
-              // case: login requires MFA step
-              setStep(2);
-              setIsLoading(false);
-              return;
+            if (!email || !password) {
+                return;
             }
-    
-            // case: login does not require MFA step
-            router.push(`/dashboard/${localStorage.getItem('projectData.id')}`);
-          }
-    
+
+            setIsLoading(true);
+            const queryParams = new URLSearchParams(location.search)
+            if (queryParams) {
+                const callbackPort = queryParams.get("callback_port")
```

Comment:
I think this line should go above the if and we should only go into it if 

```
const callbackPort = queryParams.get("callback_port")
```
has a value

Reply:
Wouldn't `queryParams` if undefined throw an error if we call `.get()` on it?

Reply:
Maybe an extra check after the line for the callback_port value instead?

Reply:
I think this should actually be fine